Jenkinsfile

```diff
@@ -1,41 +1,84 @@
-if (JENKINS_URL.contains('infra.ci.jenkins.io')) {
-  buildDockerAndPublishImage('incrementals-publisher', [automaticSemanticVersioning: true])
-  return
-}
+pipeline {
+  options {
+    timeout(time: 60, unit: 'MINUTES')
+    ansiColor('xterm')
+    disableConcurrentBuilds(abortPrevious: true)
+    buildDiscarder logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '5')
+  }
+
+  agent {
+    label 'node'
+  }
 
-if (JENKINS_URL.contains('ci.jenkins.io')) {
-  properties([
-      buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '5')),
-      disableConcurrentBuilds(),
-      disableResume()
-  ])
-  node('docker&&linux') {
-    timeout(60) {
-      ansiColor('xterm') {
-        stage('Checkout source') {
-          checkout scm
+  environment {
+    NODE_ENV = 'production'
+    TZ = "UTC"
+    NETLIFY = "true"
+  }
+
+  stages {
+    stage('Check for typos') {
+      steps {
+        sh '''typos --format json | typos-checkstyle - > checkstyle.xml || true'''
+      }
+      post {
+        always {
+          recordIssues(tools: [checkStyle(id: 'typos', name: 'Typos', pattern: 'checkstyle.xml')])
         }
-        stage('NPM Install') {
-          runDockerCommand('node:18',  'npm ci')
+      }
+    }
+
+    stage('Install Dependencies') {
+      environment {
+        NODE_ENV = 'development'
+      }
+      steps {
+        sh 'asdf install'
+        sh 'npm ci'
+      }
+    }
+
+    stage('Lint') {
+      steps {
+        sh '''
+          npx eslint --format checkstyle . > eslint-results.json
+        '''
+      }
+      post {
+        always {
+          recordIssues(tools: [
+              esLint(pattern: 'eslint-results.json'),
+          ])
         }
-        stage('Lint and Test') {
-          runDockerCommand('node:18',  'npm run test')
+      }
+    }
+
+    stage('Test') {
+      steps {
+        sh 'npm run test --if-present'
+      }
+    }
+
+    stage('Build') {
+      steps {
+        sh 'npm run build --if-present'
+      }
+    }
+
+    stage('Release') {
+      when {
+        allOf {
```

Comment:
No need for the branch check : https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildDockerAndPublishImage.groovy holds the conditional logic to publish only on the main branch (or on a tag).

Removing the branch condition also enable the hadolint/docker build in prs (instead of waiting for a release to realize the dockerfile is broken 😅)

Reply:
can you make the change? I snuck this in before i started working, i shouldn't really touch it again till tonight.

Reply:
@halkeye done in https://github.com/jenkins-infra/incrementals-publisher/pull/47